src/index.js

```diff
@@ -19,13 +19,42 @@
 // then copied from proposal-frozen-realms deep-freeze.js
 // then copied from SES/src/bundle/deepFreeze.js
 
-function makeHardener(initialFringe) {
+/**
+ * @typedef HardenerOptions
+ * @type {object}
+ * @property {WeakSet=} fringeSet WeakSet to use for the fringeSet
+ * @property {Function=} naivePrepareObject Call with object before hardening
+ */
+
+/**
+ * Create a `harden` function.
+ * 
+ * @param {Iterable} initialFringe Objects considered already hardened
+ * @param {HardenerOptions=} opts Options for creation
+ */
+function makeHardener(initialFringe, opts={}) {
   const { freeze, getOwnPropertyDescriptors, getPrototypeOf } = Object;
   const { ownKeys } = Reflect;
+
+  if (opts.fringeSet) {
```

Comment:
This code tests for the existence of opts.fringeSet three time (lines 39, 49, 51). The opts object comes from the caller, and so may be an object that gives inconsistent answers. An old defensive programming practice (that IIRC @dtribble and I coined when we were developing in Joule) is

"If you don't want inconsistent answers, only ask orthogonal questions."

As stated, this ideal is often unachievable. But it's a good ideal to approach. Here, I'd extract `opts.fringeSet` into a variable once. Do one test of that variable --- not because there's any further instability danger, but because the code will be clearer. And then in the else path overwrite it with `new WeakSet(initialFringe)`.

Reply:
On further reflection, I don't like the redundancy between `initialFringe` and `opt.fringeSet`. Perhaps it would be better if the one mandatory argument were a set instead of an iterable?

Note that any such change would be a compat break, which we now need to worry about. So we need to discuss this.

Reply:
For less compatibility breakage, if the mandatory argument is a WeakSet, then it could be used directly.  Otherwise, it would be an iterable that initializes a newly-allocated WeakSet.

Can we reliably detect this situation?

Reply:
I don't think we should support sniffing. These two ways of thinking about the argument are quite different. The existing `initialFringe` argument is only read. The `fringeSet` itself is updated. Switching the type is too subtle a way to switch this level of semantics.

Further, such sniffing could break currently correct code. A set is already an iterable. A set can already be provided as first argument, in which case it is currently safely copied into the internal `fringeSet`.

Rather, should we introduce a compat breaking change, and use semver to alert our clients of the change? Is it too late?

For now, I propose that you address my first comment and not the second. We should file an issue reiterating the second comment and this thread, but address it after we merge this PR.